hq/app/service/CacheService.scala

```diff
@@ -0,0 +1,58 @@
+package service
+
+import akka.actor.ActorSystem
+import com.gu.Box
+import config.Config
+import model._
+import play.api.Configuration
+import rx.lang.scala.Observable
+import utils.attempt.{Attempt, FailedAttempt}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class CacheService(config: Configuration) {
+  private val accounts = Config.getAwsAccounts(config)
+
+  def update[T](actorSystem: ActorSystem)(awsWorkerFn: (List[AwsAccount]) => Attempt[T])(boxUpdaterFn: Attempt[T] => Attempt[Unit])(implicit ec: ExecutionContext) = {
+    val cacheInterval = Config.getAwsApiCacheInterval(config)
+    val subscription = Observable.interval(10.millisecond, cacheInterval).doOnEach { _ =>
+      boxUpdaterFn(awsWorkerFn(accounts))
+    }.subscribe
+
+    actorSystem.registerOnTermination(
+      subscription.unsubscribe()
+    )
```

Comment:
It is always difficult to walk the line between DRY and code coupling; in this case, I would prefer an approach where the caches use their own update functions. This would improve the maintainability and readability, and simplify the method since we no longer have to pass the necessary method into it.

We ran into difficulties when trying this out and then debugging the cache services (e.g. which call to update() failed?). I also believe that the

If we want to update the caches at different intervals, that would require further abstraction.